value/list.go

```diff
@@ -0,0 +1,80 @@
+/*
+Copyright 2019 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package value
+
+// List represents a list object.
+type List interface {
+	// Length returns how many items can be found in the map.
+	Length() int
+	// At returns the item at the given position in the map. It will
+	// panic if the index is out of range.
+	At(int) Value
+}
+
+// Equals compares two lists lexically.
+func ListEquals(lhs, rhs List) bool {
+	if lhs.Length() != rhs.Length() {
+		return false
+	}
+
+	for i := 0; i < lhs.Length(); i++ {
+		lv := lhs.At(i)
+		rv := rhs.At(i)
+		if !Equals(lv, rv) {
+			lv.Recycle()
+			rv.Recycle()
+			return false
+		}
+		lv.Recycle()
+		rv.Recycle()
+	}
+	return true
+}
+
+// Less compares two lists lexically.
```

Comment:
If you don't like the idea, you should fix the comments to match the function names :)